core/src/main/scala/io/finch/DecodeEntity.scala

```diff
@@ -54,3 +40,22 @@ object DecodeEntity {
     else Try(UUID.fromString(s))
   )
 }
+
+trait LowPriorityDecode {
+
+  /**
+    * Creates an [[DecodeEntity]] instance from a given function `String => Try[A]`.
+    */
+  def instance[A](fn: String => Try[A]): DecodeEntity[A] = new DecodeEntity[A] {
+    def apply(s: String): Try[A] = fn(s)
+  }
+
+  /**
+    * Creates a [[Decode]] from [[shapeless.Generic]] for single value case classes.
+    */
+  implicit def decodeFromGeneric[A, H <: HList, E](implicit
```

Comment:
My guess is if we remove this, things like `param[String].as[Foo]` will stop working.

Reply:
Yes. Also `param[Foo]` won't work anymore.

Reply:
I'm less worried about `param[Foo]` (that said, I'm not 100% sure we need/want to support that), but we definitely need / want to keep.